src/operations/execute_operation.ts

```diff
@@ -87,8 +89,7 @@ export function executeOperation<
         owner = Symbol();
         session = topology.startSession({ owner, explicit: false });
       } else if (session.hasEnded) {
-        // TODO(NODE-3405): Change this out for MongoExpiredSessionError
-        return cb(new MongoDriverError('Use of expired sessions is not permitted'));
+        return cb(new MongoExpiredSessionError('Use of expired sessions is not permitted'));
       } else if (session.snapshotEnabled && !topology.capabilities.supportsSnapshotReads) {
         return cb(new MongoDriverError('Snapshot reads require MongoDB 5.0 or later'));
```

Comment:
I think this might have been missed as an api error because snapshots were implemented while the other PR was being implemented, can we add the correct compatibility error since we are touching this file?